packages/api-fetch/src/middlewares/preloading.js

```diff
@@ -32,8 +32,9 @@ const createPreloadingMiddleware = ( preloadedData ) => ( options, next ) => {
 		const method = options.method || 'GET';
 		const path = getStablePath( options.path );
 
-		if ( 'GET' === method && preloadedData[ path ] ) {
-			return Promise.resolve( preloadedData[ path ].body );
+		if ( preloadedData[ method ] && preloadedData[ method ][ path ] ) {
+			const response = 'OPTIONS' === method ? preloadedData[ method ][ path ] : preloadedData[ method ][ path ].body;
```

Comment:
**Coment from old PR**
https://github.com/WordPress/gutenberg/pull/4155#discussion_r201326209

> We should have a unit test for this. I'm also not sure why we ought to need to special-case OPTIONS requests. Who's to say we don't want other headers from the request? Just seems like we're making inconsistency for not an entirely valid reason.

Reply:
I can see that this would be quite a big refactor, so maybe worth splitting into a separate PR?